Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -74,22 +74,26 @@ public Stream Fetch(string key)
 
                     if (dataStream != null)
                     {
-                        _zipFileCache.AddOrUpdate(filename,
-                            x =>
+                        Lazy<CachedZipFile> existingEntry;
+                        if (!_zipFileCache.TryGetValue(filename, out existingEntry))
+                        {
+                            try
                             {
                                 var newItem = new CachedZipFile(ZipFile.Read(dataStream), filename);
                                 stream = CreateStream(newItem.ZipFile, entryName);
-                                return newItem;
-                            },
-                            (x, existingEntry) =>
+                                _zipFileCache.TryAdd(filename, new Lazy<CachedZipFile>(() => newItem));
```

Comment:
@andrewhart098 do you know if we needed/relied on the previous behavior where it recreates the stream part each time?